apis/compute/v1beta1/backendservice_reference.go

```diff
@@ -0,0 +1,133 @@
+	referenceContext := refsv1beta1.ReferenceContext{IsDirectOnly: false, TargetField: "status.selfLink"}
+	// Get value from spec.ComputeBackendServiceRef.external
+	if r.External != "" {
+		if r.Name != "" {
+			return "", fmt.Errorf("cannot specify both name and external on reference")
+		}
+		if referenceContext.IsDirectOnly {
+			if _, err := ParseBackendServiceExternal(r.External); err != nil {
+				return "", err
+			}
+		}
+		// To ensure backward compatibility for existing users, we do not enforce external format
+		// for non-direct resources
+		return r.External, nil
+	}
+
+	if r.Name == "" {
+		return "", fmt.Errorf("must specify either name or external on reference")
+	}
+
+	// Get value from the Config Connector object
+	if r.Namespace == "" {
+		r.Namespace = otherNamespace
+	}
+	key := types.NamespacedName{Name: r.Name, Namespace: r.Namespace}
+
+	u, err := refsv1beta1.ResolveResourceName(ctx, reader, key, ComputeBackendServiceGVK)
+	if err != nil {
+		return "", err
+	}
+
+	actualExternalRef, found, err := unstructured.NestedString(u.Object, "status", "externalRef")
+	if err != nil {
+		return "", fmt.Errorf("error getting status.externalRef for %s %s/%s: %w", u.GetKind(), u.GetNamespace(), u.GetName(), err)
+	}
+	// If object is DirectOnly, it is created by direct controller.
+	// Get value from status.externalRef, which is the most trustworthy source.
+	if referenceContext.IsDirectOnly {
+		if !found || actualExternalRef == "" {
+			return "", fmt.Errorf("status.externalRef is required for DirectOnly resources, but is missing or empty for %s %s/%s", u.GetKind(), u.GetNamespace(), u.GetName())
+		}
+		r.External = actualExternalRef
+	}
+
+	// If object not DirectOnly, it can be created by either direct controller or legacy controller, depends on user's settings.
+	// If status.externalRef does not exist, it's created by legacy controller. Get values from target field.
+	if !found {
+		targetField := referenceContext.TargetField
+		tokens := strings.Split(targetField, ".")
+		targetField, found, err := unstructured.NestedString(u.Object, tokens...)
+		if err != nil {
+			return "", fmt.Errorf("error getting target field %s for %s %s/%s: %w", targetField, u.GetKind(), u.GetNamespace(), u.GetName(), err)
+		}
+		if !found || targetField == "" {
+			return "", fmt.Errorf("target field %s is required but is missing or empty for %s %s/%s", targetField, u.GetKind(), u.GetNamespace(), u.GetName())
+		}
+		r.External = targetField
+	} else {
+		// If status.externalRef exists, it's created by direct controller. Get value from status.externalRef.
+		r.External = actualExternalRef
+	}
+	return r.External, nil
+}
+
+func ParseBackendServiceExternal(external string) (identity *BackendServiceIdentity, err error) {
+	external = strings.TrimPrefix(external, "/")
+	tokens := strings.Split(external, "/")
+	if len(tokens) == 5 && tokens[0] == "projects" && tokens[3] == "backendServices" {
+		return &BackendServiceIdentity{
```

Comment:
One thought: maybe we could rename <ResourceIdentity> to <ResourceExternal>? I feel "Identity" is just the same thing as "ExternalRef", so that we don't introduce many new terms to people.